### DISCUSSION TITLE
How to force a dynamic checkbox group to always return a list on submit

### DISCUSSION THREAD

**Discussion by GV94** [react-hook-form/react-hook-form#3693] [General] [answered]
I have a dynamic group of checkboxes in a form

Example
```html
  <form>
     {listOfStuff.map(thing => <input type="checkbox" name="stuff" value={thing.name} />
  </form>
```

I want to handle this data as an array, but if I only render one checkbox (`listOfStuff.length === 1`), the value is string, not a an array of strings. Is there a way to enforce this without having to render an invisible checkbox?

**Comment by bluebill1049** [comment, marked as answer]
I think it's probably easier to build a controlled input and wrapped with `Controller`, then you can easier to manage your data type.

**Reply by GV94** [reply]
Alright, thank you for the response.
